### PR TITLE
Auto-mirroring support in RTL mode

### PR DESCRIPTION
**Pull request by 1zaman**
`IconDrawable` has been modified to support auto-mirroring by default if supported by the icon. I have also added an `IconImageView` widget with custom attributes at edx/edx-app-android@d2d4fc6084ba63b3d5bda40a7c2b45258a16b778 to support XML definitions, but in order to add that here this project would need to be changed to an Android library module, which I don't have any experience with and probably requires integrating with Gradle. If you or someone else manages to make this transition, I'll be happy to include that commit here as well.

### CODE REVIEW

**Comment by JoanZapata** [comment]
Thanks for contributing your changes, however I don't think *all* icons should be mirrored to support RTL. Logos, for example, should not. I'm closing this PR for now, feel free to reopen if you can take the time to only mirror the few icons that make sense (arrows I guess). :)

**Comment by 1zaman** [comment]
I have already defined the icons that don’t support RTL as non-mirrorable. Please take a look at c5fb9b7823dc99781f6e39964b68f5ec94de2a24

**Comment by JoanZapata** [comment]
Oops! My bad, sorry, checked a little bit too fast. It's perfect thank you. :)

**Comment by 1zaman** [comment]
Thanks for merging. Do you have any plans of converting it into an Android
module?

File: android-iconify/src/main/java/com/joanzapata/android/iconify/IconDrawable.java [hunk 1]
```diff
@@ -208,4 +232,37 @@ public void setStyle(Paint.Style style) {
...
+    @Override
+    public final boolean isAutoMirrored() {
+        return autoMirrored;
+    }
+
+    @TargetApi(KITKAT)
+    private boolean needMirroring() {
+        if (isAutoMirrored()) {
+            // Since getLayoutDirection() is hidden, we will try to
+            // get the layout direction from the View, which we will,
```
Comment (1zaman) [inline android-iconify/src/main/java/com/joanzapata/android/iconify/IconDrawable.java]: Taking a second look through the diff, I see that the comma at the end of this line is incorrect. You can remove that if you want.

**Comment by JoanZapata** [comment]
At the time I created the lib a `jar` made the build faster than an `apklib` (or `aar`).
I don't know if that's still the case, but I don't have any plan.
Why?

**Comment by 1zaman** [comment]
If this library is converted to an AAR, then we can include custom attribute definitions, such as the `iconName` enums for my `IconImageView` at edx/edx-app-android@d2d4fc6084ba63b3d5bda40a7c2b45258a16b778. Lacking this option, the `IconDrawable` cannot be used from XML layouts, which makes it more cumbersome to use globally.

**Comment by 1zaman** [comment]
While this seems to have been reverted during Font Awesome 4.4.0 upgrade, I have updated my branch to merge all the latest changes on develop, including support for new icon sets and `CustomTypefaceSpan`. I might or might not keep it updated in the future, as currently I only use Font Awesome icons, and don't really need to keep that updated either.

**Comment by JoanZapata** [comment]
Yes sorry I reverted it because it makes updating the font libraries way too long and I can't afford to spend that much time on it.

**Comment by 1zaman** [comment]
That's understandable. You *could* retain the functionality though, and
just disable it for your own modules (or enable it for specific fonts that
aren't updated often or don't have large updates). That would allow me to
use your main module in my project (after my other pull requests are merged
😉), while using my own font module implementation. Other people might also
benefit from this feature.

**Pull request merged**